File: exercises_03/18.java
/* 
	Program that:
		Determine si un número es primo o no.
*/

import java.util.Scanner;

public class Main {
	public static void main (String[] args) {
		Scanner userInput = new Scanner(System.in);

		int number = userInput.nextInt();

		for (int i = 1; i < number; i++) {
			if ( (number % i) == 0 && ((number != i) && (i != 1) )) {
				System.out.printf("No Primo\n");
				return;
			}
		}

		System.out.printf("Primo\n");
		userInput.close();
		return;
	}
}

File: exercises_04/18.cpp
/*
 * =====================================================================================
 *       Filename:  17.cpp
 *    Description:  Program that: Escriba una función que determine el número de cifras de un número en base b sin usar librerías de matemáticas.
 *        Version:  1.0
 *       Compiler:  gcc
 * =====================================================================================
 */

#include <stdio.h>

int get_number_digits_in_base (int number, int base) {
    int number_digits = 0;
    int aux_number = number;

    while (aux_number > 0) {
        aux_number /= base;
        number_digits++;
    }

    return number_digits;
}

int main (void) {
    int number = 0;
    int base = 0;
    int num_digits = 0;

    fscanf(stdin, "%d", &number);
    fscanf(stdin, "%d", &base);

    num_digits = get_number_digits_in_base(number, base);

    fprintf(stdout, "%d\n", num_digits);

    return 0;
}
